mlir-tensorrt/compiler/lib/Compiler/TensorRTToExecutable/Passes.cpp

```diff
@@ -63,15 +63,29 @@ createOutlinedFunc(RewriterBase &rewriter, Location loc, Operation *module,
   FunctionType type =
       FunctionType::get(rewriter.getContext(), funcArgTypes, funcResultTypes);
   auto outlinedFunc = func::FuncOp::create(loc, nameBase, type, {});
+
+  StringRef tensorrtShapeBoundsAttrName = mlir::tensorrt::TensorRTDialect::getShapeProfileArgAttrName();
+
+  // TODO (pranavm): Will there only ever be one outer FuncOp? How do we make this more robust?
+  func::FuncOp outerFunc;
+  for (auto func : outerModule.getOps<func::FuncOp>()) {
+      outerFunc = func;
+      break;
+  }
+  for (unsigned idx = 0; idx < outlinedFunc.getNumArguments(); idx++) {
+    const auto profile = outerFunc.getArgAttrOfType<tensorrt::ShapeProfileAttr>(idx, tensorrtShapeBoundsAttrName);
```

Comment:
The arguments of the outlined function may not be the same or ordered the same as the function which contained the cluster.